troposphere/codebuild.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2016, Mark Peek <mark@peek.org>
+# All rights reserved.
+#
+# See LICENSE file for full license.
+
+from . import AWSObject, AWSProperty, Tags
+from .validators import integer
+
+
+class Artifacts(AWSProperty):
+    props = {
+        'Location': (basestring, False),
+        'Name': (basestring, False),
+        'NameSpaceType': (basestring, False),
+        'Packaging': (basestring, False),
+        'Path': (basestring, False),
+        'Type': (basestring, True),
+    }
+
+    def validate(self):
+        valid_types = [
+            'CODEPIPELINE',
+            'NO_ARTIFACTS',
+            'S3',
+        ]
+        artifact_type = self.properties.get('Type')
+        if artifact_type not in valid_types:
+            raise ValueError('Artifacts Type: must be one of %s' %
+                             ','.join(valid_types))
+
+        if artifact_type == 'S3':
+            for required_property in ['Name', 'Location']:
+                if not self.properties.get(required_property):
+                    raise ValueError(
+                        'Artifacts Type S3: requires %s to be set' %
+                        required_property
+                    )
+
+
+class EnvironmentVariable(AWSProperty):
+    props = {
+        'Name': (basestring, True),
+        'Value': (basestring, True),
+    }
+
+
+class Environment(AWSProperty):
+    props = {
+        'ComputerType': (basestring, True),
```

Comment:
I think this should be 'ComputeType'

http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codebuild-project-environment.html#cfn-codebuild-project-environment-computetype